src/main/java/frc/robot/subsystems/SwerveSubsystem.java

```diff
@@ -0,0 +1,136 @@
+import com.ctre.phoenix6.mechanisms.swerve.SwerveModule;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveModuleConstants;
+import com.ctre.phoenix6.mechanisms.swerve.SwerveRequest;
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.commands.PathPlannerAuto;
+import com.pathplanner.lib.util.HolonomicPathFollowerConfig;
+import com.pathplanner.lib.util.PIDConstants;
+import com.pathplanner.lib.util.ReplanningConfig;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import frc.robot.Constants;
+import java.util.function.Supplier;
+
+/**
+ * Class that extends the Phoenix SwerveDrivetrain class and implements subsystem so it can be used
+ * in command-based projects easily.
+ */
+public class SwerveSubsystem extends SwerveDrivetrain implements Subsystem {
+  // instance of SwerveSubsystem
+  private static SwerveSubsystem instance;
+
+  // Constructor allows for custom odometry update frequency
+  public SwerveSubsystem(
+      SwerveDrivetrainConstants driveTrainConstants,
+      double OdometryUpdateFrequency,
+      SwerveModuleConstants... modules) {
+
+    // Sets the drivetrain constants, odometry update frequency and constants for
+    // the swerve modules
+    super(driveTrainConstants, OdometryUpdateFrequency, modules);
+
+    // Sets the supply current limits for the swerve modules (for driving and
+    // turning)
+    CurrentLimitsConfigs driveCurrentLimits =
+        new CurrentLimitsConfigs()
+            .withSupplyCurrentLimit(Constants.Swerve.kDriveSupplyCurrentLimit)
+            .withSupplyCurrentLimitEnable(true);
+
+    CurrentLimitsConfigs turningCurrentLimits =
+        new CurrentLimitsConfigs()
+            .withSupplyCurrentLimit(Constants.Swerve.kTurningSupplyCurrentLimit)
+            .withSupplyCurrentLimitEnable(true);
+
+    for (SwerveModule module : Modules) {
+      module.getDriveMotor().getConfigurator().apply(driveCurrentLimits);
+      module.getSteerMotor().getConfigurator().apply(turningCurrentLimits);
+    }
+
+    // Configures the holonomic auto builder
+    configurePathPlanner();
+  }
+
+  // Constructor for default odometry update frequency
+  public SwerveSubsystem(
+      SwerveDrivetrainConstants driveTrainConstants, SwerveModuleConstants... modules) {
+    this(driveTrainConstants, 0, modules);
+  }
+
+  public static SwerveSubsystem getInstance() {
+    if (instance == null) {
+      instance =
+          new SwerveSubsystem(
+              Constants.Swerve.DrivetrainConstants,
+              Constants.Swerve.FrontLeft,
+              Constants.Swerve.FrontRight,
+              Constants.Swerve.BackLeft,
+              Constants.Swerve.BackRight);
+    }
+    return instance;
+  }
+
+  // Applies swerve request to drivetrain
+  public Command applyRequest(Supplier<SwerveRequest> requestSupplier) {
```

Comment:
the naming on this is a little misleading... it's cause a bug before because it doesn't actually apply the request as the name suggests but returns a command that would run the request.

Reply:
renamed to `getSwerveRequest`